### DISCUSSION TITLE
Real scrolling for Radiotext

### DISCUSSION THREAD

**Discussion by andimik** [PE5PVB/TEF6686_ESP32#68] [Ideas]
Currently Radiotext will immediately start again from the 1st character after the last was displayed on the right. In some cases the text is hard to read (it's like ping-pong).

So I have two ideas for Radiotext.

- Real scrolling (like in a wheel) including the leading text `RT: ` and some ending spaces
- add new option to show only full received RT (which will only work for error-free signals of course)

This behavior is used by my old DENON Receiver, btw.

**Comment by PE5PVB** [comment]
I've added some extra spaces. Please check. Problem of retrieving full RT, when only one cycle is missing, you have nothing to show. Radiotext is send in blocks of 4 bytes, so when you have low signal, it can take a long while to receive full RT, if the A/B flag is not triggered during this.

**Comment by andimik** [comment]
Please add a prefix like `Radiotext:`

Because when it's scrolling, the first character only will be shown within a fraction of a second.

And secondly, it's scrolling during reception. So every new received byte will scroll to the left and disappear ...

So let me propose that Radiotext shall get a delay instead of retrieving full RT.

**Comment by andimik** [comment]
I suggest to use a much higher value for xPos (for example 60). Then the text will start scrolling nearly in the middle of the field and you won't miss the first characters in the ticker.

**Reply by PE5PVB** [reply]
I will play with it. I was thinking about a short pause at the start and end of scrolling.

**Comment by andimik** [comment]
yes, true, I also wanted to implement it, but I could not find a parameter for the start delay.

**Comment by PE5PVB** [comment]
You can use a counter with millis() for that.

**Reply by andimik** [reply]
No, changing the counter in ms will speed up/down the scrolling.

**Reply by PE5PVB** [reply]
Yes, but you can trigger another counter at start and end of the scroll just to hold it for a few milliseconds

**Comment by andimik** [comment]
Well, when `Show underscore in RDS` is **off**, then this behavior is not so bad than otherwise. 

Could you ignore underscores (= placeholders) in the ticker, so it will only scroll when a character is other than `[SPACE]` or `_`.

**Reply by PE5PVB** [reply]
No, not possible. It's just a counter of the total string.

**Reply by andimik** [reply]
Damn. But my idea was good :-)

**Comment by PE5PVB** [comment]
I created a short pause before scrolling starts, also scrolling is much smoother now.
